src/components/CommentsCard.jsx

```diff
@@ -6,22 +6,36 @@ import Comment from "./Comment";
 import CommentForm from "./CommentForm";
 
 const CommentsCard = () => {
+  const loggedInUser = "grumpy19";
   const { article_id } = useParams();
   const [comments, setComments] = useState([]);
+
   const addComment = (text) => {
     axios.post(
       `https://nc-news-czlb.onrender.com/api/articles/${article_id}/comments`,
-      { username: "grumpy19", body: text }
+      { username: loggedInUser, body: text }
     );
   };
+
+  const deleteComment = (comment.comment_id) => {
+    axios
+      .delete(
+        `https://nc-news-czlb.onrender.com/api/comments/${comment.comment_id}`
+      )
+      .then(
+        setComments((currentComments) => [
+          currentComments.splice(indexOf(comment.comment_id), 1),
+        ])
+      );
+  };
```

Comment:
The way that you're updating the comments stat using splice and indexOf isnt quite correct 

- indexOf needs to be attached to an array e.g. `array.indexOf` instead of how you've used it here
- splice returns the deleted items, not the new array. We'd want to update state without modifying the existing state (pure functions)
- You're using `[...]` around the splice which would end up setting comments to an array containing the deleted items instead of a new array without the deleted items

I'd recommend a refactor where you replace this with a .filter on the current comments, it's a much easier way to achieve what we want!